src/mediator.js

```diff
@@ -0,0 +1,5 @@
+import Vue from 'vue';
```

Comment:
this dummy component functions like a mediator pattern, giving us a central hub to dispatch events between components. Technically, this would allow components to communicate up and down and side to side in a tree. In practice, I think that's going to get messy awful darn quick. In this case, i've used it to quickly bubble events up to the root component from relatively deeply nested components, but downward communication is handled entirely through props. I don't know if this pattern will work well at an extremely complex scale (i bet not), but it actually wasn't half bad at this fairly low level of complexity.

Reply:
Seems to accomplish the same pattern as exposing a global state object to communicate with / subscribe to / emit actions to - ala Redux or MST. You're just rolling your own mechanism. :)

Reply:
yes, basically. for the really limited scope of this application, it was easier than trying to set up a whole flux store. but as these components live and grow, it would be easy to refactor to that later when the complexity warranted. I believe that's why this is the suggested approach for simple apps